pandas/_libs/tslibs/strptime.pyx

```diff
@@ -156,212 +157,203 @@ def array_strptime(
 
     for i in range(n):
         val = values[i]
-        if isinstance(val, str):
-            if val in nat_strings:
-                iresult[i] = NPY_NAT
-                continue
-        elif checknull_with_nat_and_na(val):
-            iresult[i] = NPY_NAT
-            continue
-        elif PyDateTime_Check(val):
-            if val.tzinfo is not None:
-                found_tz = True
-            else:
-                found_naive = True
-            tz_out = convert_timezone(
-                val.tzinfo,
-                tz_out,
-                found_naive,
-                found_tz,
-                utc,
-            )
-            if isinstance(val, _Timestamp):
-                iresult[i] = val.tz_localize(None).as_unit("ns").value
-            else:
-                iresult[i] = pydatetime_to_dt64(val.replace(tzinfo=None), &dts)
-                check_dts_bounds(&dts)
-            result_timezone[i] = val.tzinfo
-            continue
-        elif PyDate_Check(val):
-            iresult[i] = pydate_to_dt64(val, &dts)
-            check_dts_bounds(&dts)
-            continue
-        elif is_datetime64_object(val):
-            iresult[i] = get_datetime64_nanos(val, NPY_FR_ns)
-            continue
-        elif (
-                (is_integer_object(val) or is_float_object(val))
-                and (val != val or val == NPY_NAT)
-        ):
-            iresult[i] = NPY_NAT
-            continue
-        else:
-            val = str(val)
-
-        # exact matching
-        if exact:
-            found = format_regex.match(val)
-            if not found:
-                if is_coerce:
-                    iresult[i] = NPY_NAT
-                    continue
-                raise ValueError(f"time data '{val}' does not match "
-                                 f"format '{fmt}' (match)")
-            if len(val) != found.end():
-                if is_coerce:
-                    iresult[i] = NPY_NAT
-                    continue
-                raise ValueError(f"unconverted data remains: {val[found.end():]}")
-
-        # search
-        else:
-            found = format_regex.search(val)
-            if not found:
-                if is_coerce:
+        try:
```

Comment:
is the code inside this try/except just indenting whats currently there or are there other changes?

Reply:
it's basically just that, but I also removed some
```
                if is_coerce:
                    iresult[i] = NPY_NAT
                    continue
```
as that already happens in the outer try-except

If you hide whitespace it'll be easier to see - you can also go to https://github.com/pandas-dev/pandas/pull/50265/files?diff=unified&w=1 (notice the `?diff=unified&w=1` at the end) to see it

Reply:
Generally I'm not a big fan of try/excepts with a huge block of code inside, but I guess since the failures modes here should be made consistent with the broader `to_datetime(errors=)` keyword I suppose this is fine. 

Question, sometime we are issues some `ValueError`s internally in this function that will be caught by this try except e.g. 

```
            if iso_year != -1:
                if iso_week == -1 or weekday == -1:
                    raise ValueError("ISO year directive '%G' must be used with "
                                     "the ISO week directive '%V' and a weekday "
                                     "directive '%A', '%a', '%w', or '%u'.")
```

Is this necessarily a "parsing" error that should be influenced by the errors keyword?

Reply:
I can rewrite it such that each `check_dts_bounds` is within its own try-except - maybe that's better actually, thanks

I'll get back to you on that ValueError

Reply:
OK, refactored

That's a _really_ good point about the "ISO year directive" error, thanks, good catch! So, I think that one should always raise. At least, that seems to be the current intention, there's no `if is_coerce` check there like there is for some other errors, such as

https://github.com/pandas-dev/pandas/blob/a5cbd1e52e9078637c899167f40a29b4bc8901b9/pandas/_libs/tslibs/strptime.pyx#L378-L382

So, I've parametrised `test_error_iso_week_year` too over `errors` to check that this one actually raises

Reply:
I am not actually opposed to the large try/except block if we expect ValueError and OutofBounds Errors to be affected by `errors`. If other excepts in this functions can be factored out (like the iso year directive one) and others then it would help ensure errors is consistently handled

Reply:
OK thanks - have factored out the iso year directive check in #50309, so I've gone back to having an outer try-except block (so again, "hide-whitespace" will come in handy when reviewing)